// @banneratriads/button/dist/collection/components/button/button.js
import { h } from "@stencil/core";
export class AtriaButtonComponent {
    constructor() {
        // Props equivalent to Angular @Input()
        this.buttonId = "";
        this.buttonText = "";
        this.buttonType = 'primary';
        this.buttonIconLeft = "";
        this.buttonIconRight = "";
        this.disabled = false;
        this.loading = false;
        this.active = false;
        this.width = "";
        this.buttonRole = 'button';
        // Click Handler
        this.onClick = () => {
            if (!this.disabled && !this.loading) {
                this.btnClick.emit({
                    id: this.buttonId,
                    text: this.buttonText,
                    type: this.buttonType,
                    active: this.active
                });
            }
        };
    }
    render() {
        const classes = [
            'bh-btn',
            this.buttonType,
            this.loading ? 'bh-loading' : '',
            this.disabled ? 'disabled' : '',
            this.active ? 'active' : '',
            !this.buttonText && !this.loading && this.buttonIconLeft ? 'bh-icon-only' : ''
        ]
            .filter(Boolean)
            .join(' ');
        return (h("button", { key: 'f1ccb22e88e0032e9608606bc97bf04e8ec37e35', id: this.buttonId, class: classes, style: { width: this.width }, type: this.buttonRole, onClick: this.onClick, disabled: this.disabled }, this.buttonIconLeft && !this.loading && (h("i", { key: '89282a3959bc4554a8549decea903c737ca2c498', class: `left bh-icon bh-icon-md ${this.buttonIconLeft}`, "aria-label": this.buttonText })), this.buttonText, this.buttonIconRight && !this.loading && (h("i", { key: '46386ca2945aa40612080999891641920719bbd2', class: `right bh-icon bh-icon-md ${this.buttonIconRight}`, "aria-label": this.buttonText }))));
    }
    static get is() { return "atria-button"; }
    static get encapsulation() { return "shadow"; }
    static get originalStyleUrls() {
        return {
            "$": ["button.scss"]
        };
    }
    static get styleUrls() {
        return {
            "$": ["button.css"]
        };
    }
    static get properties() {
        return {
            "buttonId": {
                "type": "string",
                "mutable": false,
                "complexType": {
                    "original": "string",
                    "resolved": "string",
                    "references": {}
                },
                "required": false,
                "optional": false,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false,
                "attribute": "button-id",
                "reflect": false,
                "defaultValue": "\"\""
            },
            "buttonText": {
                "type": "string",
                "mutable": false,
                "complexType": {
                    "original": "string",
                    "resolved": "string",
                    "references": {}
                },
                "required": false,
                "optional": false,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false,
                "attribute": "button-text",
                "reflect": false,
                "defaultValue": "\"\""
            },
            "buttonType": {
                "type": "string",
                "mutable": false,
                "complexType": {
                    "original": "string",
                    "resolved": "string",
                    "references": {}
                },
                "required": false,
                "optional": false,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false,
                "attribute": "button-type",
                "reflect": false,
                "defaultValue": "'primary'"
            },
            "buttonIconLeft": {
                "type": "string",
                "mutable": false,
                "complexType": {
                    "original": "string",
                    "resolved": "string",
                    "references": {}
                },
                "required": false,
                "optional": false,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false,
                "attribute": "button-icon-left",
                "reflect": false,
                "defaultValue": "\"\""
            },
            "buttonIconRight": {
                "type": "string",
                "mutable": false,
                "complexType": {
                    "original": "string",
                    "resolved": "string",
                    "references": {}
                },
                "required": false,
                "optional": false,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false,
                "attribute": "button-icon-right",
                "reflect": false,
                "defaultValue": "\"\""
            },
            "disabled": {
                "type": "boolean",
                "mutable": false,
                "complexType": {
                    "original": "boolean",
                    "resolved": "boolean",
                    "references": {}
                },
                "required": false,
                "optional": false,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false,
                "attribute": "disabled",
                "reflect": false,
                "defaultValue": "false"
            },
            "loading": {
                "type": "boolean",
                "mutable": false,
                "complexType": {
                    "original": "boolean",
                    "resolved": "boolean",
                    "references": {}
                },
                "required": false,
                "optional": false,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false,
                "attribute": "loading",
                "reflect": false,
                "defaultValue": "false"
            },
            "active": {
                "type": "boolean",
                "mutable": false,
                "complexType": {
                    "original": "boolean",
                    "resolved": "boolean",
                    "references": {}
                },
                "required": false,
                "optional": false,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false,
                "attribute": "active",
                "reflect": false,
                "defaultValue": "false"
            },
            "width": {
                "type": "string",
                "mutable": false,
                "complexType": {
                    "original": "string",
                    "resolved": "string",
                    "references": {}
                },
                "required": false,
                "optional": false,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false,
                "attribute": "width",
                "reflect": false,
                "defaultValue": "\"\""
            },
            "buttonRole": {
                "type": "string",
                "mutable": false,
                "complexType": {
                    "original": "string",
                    "resolved": "string",
                    "references": {}
                },
                "required": false,
                "optional": false,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "getter": false,
                "setter": false,
                "attribute": "button-role",
                "reflect": false,
                "defaultValue": "'button'"
            }
        };
    }
    static get events() {
        return [{
                "method": "btnClick",
                "name": "btnClick",
                "bubbles": true,
                "cancelable": true,
                "composed": true,
                "docs": {
                    "tags": [],
                    "text": ""
                },
                "complexType": {
                    "original": "any",
                    "resolved": "any",
                    "references": {}
                }
            }];
    }
}
