// @banneratriads/button/angular/components.ts
/* tslint:disable */
/* auto-generated angular directive proxies */
import { ChangeDetectionStrategy, ChangeDetectorRef, Component, ElementRef, EventEmitter, NgZone } from '@angular/core';

import { ProxyCmp, proxyOutputs } from './angular-component-lib/utils';

import { Components } from '@vyacobucci/button';


@ProxyCmp({
  inputs: ['active', 'buttonIconLeft', 'buttonIconRight', 'buttonId', 'buttonRole', 'buttonText', 'buttonType', 'disabled', 'loading', 'width']
})
@Component({
  selector: 'atria-button',
  changeDetection: ChangeDetectionStrategy.OnPush,
  template: '<ng-content></ng-content>',
  // eslint-disable-next-line @angular-eslint/no-inputs-metadata-property
  inputs: ['active', 'buttonIconLeft', 'buttonIconRight', 'buttonId', 'buttonRole', 'buttonText', 'buttonType', 'disabled', 'loading', 'width'],
})
export class AtriaButton {
  protected el: HTMLAtriaButtonElement;
  constructor(c: ChangeDetectorRef, r: ElementRef, protected z: NgZone) {
    c.detach();
    this.el = r.nativeElement;
    proxyOutputs(this, this.el, ['btnClick']);
  }
}


export declare interface AtriaButton extends Components.AtriaButton {

  btnClick: EventEmitter<CustomEvent<any>>;
}
